cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetServiceAuthorizationTest.java

```diff
@@ -136,17 +137,23 @@ public void execute() throws Exception {
     Assert.assertNotNull(datasetSpec);
     Assert.assertEquals("val", datasetSpec.getProperty("key"));
     // grant Bob WRITE on namespace, so he can add some datasets
-    grantAndAssertSuccess(NamespaceId.DEFAULT, BOB, ImmutableSet.of(Action.WRITE));
-    addInstanceAndAssertPrivileges(BOB, dsId1);
-    addInstanceAndAssertPrivileges(BOB, dsId2);
+    grantAndAssertSuccess(dsId1, BOB, ImmutableSet.of(Action.ADMIN));
+    grantAndAssertSuccess(dsId2, BOB, ImmutableSet.of(Action.ADMIN));
+    grantAndAssertSuccess(tableTypeId, BOB, EnumSet.of(Action.ADMIN));
+    dsFramework.addInstance(Table.class.getName(), dsId1, DatasetProperties.EMPTY);
+    dsFramework.addInstance(Table.class.getName(), dsId2, DatasetProperties.EMPTY);
     // since Bob now has some privileges on all datasets, the list API should return all datasets for him
     Assert.assertEquals(ImmutableSet.of(dsId, dsId1, dsId2),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
-    // Since Alice has some privileges on the namespace, the list API should return all datasets under the namespace
+    // Alice should only be able to see dsId, since she only has privilege on this dataset
     SecurityRequestContext.setUserId(ALICE.getName());
-    Assert.assertEquals(ImmutableSet.of(dsId, dsId1, dsId2),
+    Assert.assertEquals(ImmutableSet.of(dsId),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
 
+    // Grant privileges on other datasets to user Alice
+    grantAndAssertSuccess(dsId1, ALICE, ImmutableSet.of(Action.EXECUTE));
+    grantAndAssertSuccess(dsId2, ALICE, ImmutableSet.of(Action.EXECUTE));
+
     // Alice should only be able to delete datasets that she is the ADMIN
     dsFramework.deleteAllInstances(NamespaceId.DEFAULT);
```

Comment:
According to https://issues.cask.co/browse/CDAP-6998, I think it is because we only delete datasets that the user has ADMIN privileges on, we should discuss about what is the right behavior to delete all the entities.